File: kek6.cpp
#include <iostream>
using namespace std;
int main()
{
    int fib1 = 0;
    int fib2 = 1;
    double x;
    double a;
    double fib;
    std::cin >> a;
    while (fib2 <= a){
        fib = fib1 + fib2;
        fib1 = fib2;
        fib2 = fib;
        x = fib1+fib2;
    }
    if (a == 2)
    {
        cout << fib2 << endl;
    }
    else {
        
    
    cout << x << endl;
    }
}

File: kek6.rs
use std::io::{self, Read};
fn main() -> io::Result<()>{
  let mut a = String::new();
  let fib1 = 0;
  let fib2 = 1;
  while fib2 <= a{
    fib = fib1 + fib2;
    fib1 = fib2;
    fib2 = fib;
  }
  let n = fib1 + fib2;
  writeln!(n);
  }